services/api/src/resources/routes/helpers.ts

```diff
@@ -0,0 +1,184 @@
+import { Pool } from 'mariadb';
+import { query } from '../../util/db';
+import { Sql } from './sql';
+
+export const Helpers = (sqlClientPool: Pool) => {
+  const removeAllRoutesFromEnvironment = async (environmentId: number) => {
+    await query(
+      sqlClientPool,
+      Sql.removeAllRoutesFromEnvironment(environmentId)
+    )
+  }
+  const deleteAutogeneratedRoutesForEnvironment = async (environmentId: number) => {
+    await query(
+      sqlClientPool,
+      Sql.deleteAutogeneratedRoutesForEnvironment(environmentId)
+    )
+  }
+  const deleteLagoonYAMLRoutesForEnvironment = async (environmentId: number) => {
+    await query(
+      sqlClientPool,
+      Sql.deleteLagoonYAMLRoutesForEnvironment(environmentId)
+    )
+  }
+  const removeRouteFromEnvironment = async (domain: string, environmentId: number) => {
+    const routes = await query(
+      sqlClientPool,
+      Sql.selectRoutesByDomainAndEnvironmentID(domain, environmentId)
+    )
+    if (routes.length == 0) {
+      throw new Error(`Route doesn't exist on this environment`);
+    }
+    const route = routes[0]
+    await query(
+      sqlClientPool,
+      Sql.removeRouteFromEnvironment(route.id)
+    )
+    return route.id
+  };
+  const addRouteAnnotation = async (routeId: number, key: string, value: string) => {
+    const existingAnnotations = await query(sqlClientPool, Sql.selectRouteAnnotationsByRouteID(routeId))
+    const exists = existingAnnotations.some(
+      (a) => a.key === key && a.value === value
+    );
+    if (exists) {
+      throw new Error(`Annotation already exists on route`);
+    } else {
+      const combinedAnnotationCount = existingAnnotations.length + 1
+      // arbitrary limit of 10 annotations, maybe this should be less?
+      // would prefer that annotations done this way weren't a thing, but here we are
+      if (combinedAnnotationCount >= 10) {
+        throw Error(`Limit of 10 annotations per route`)
+      }
+      await query(
+        sqlClientPool,
+        Sql.insertRouteAnnotation({routeId, key, value})
+      )
+    }
+  }
+  const addRouteAnnotations = async (routeId: number, annotations: RouteAnnotations) => {
+    const existingAnnotations = await query(sqlClientPool, Sql.selectRouteAnnotationsByRouteID(routeId))
+    for(const annotation of annotations) {
+      const exists = existingAnnotations.some(
+        (a) => a.key === annotation.key && a.value === annotation.value
+      );
+      if (exists) {
+        throw new Error(`Annotation already exists on route`);
+      }
+      const combinedAnnotationCount = existingAnnotations.length + annotations.length
```

Comment:
in fact, perhaps just centralizing this check as a function makes sense, that'll unify all the count logic.

This shouldn't be a blocker for release, but would be good to change.